demos/python_demos/common/performance_metrics.py

```diff
@@ -0,0 +1,86 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from time import perf_counter
+import cv2
+from helpers import put_highlighted_text
+
+
+class Sum:
+    def __init__(self):
+        self.latency = 0.0
+        self.time = 0.0
+        self.frame_count = 0
+
+
+class PerformanceMetrics:
+    def __init__(self, time_window=1.0):    # defines the length of the timespan used for 'current fps' calculation,
+        self.time_window_size = time_window # set to '1 second' by default
+        self.last_moving_sum = Sum()
+        self.current_moving_sum = Sum()
+        self.total_sum = Sum()
+        self.last_update_time = perf_counter()
+    
+    def update(self, last_request_start_time):
+        current_time = perf_counter()
+
+        last_request_latency = current_time - last_request_start_time
+        self.current_moving_sum.latency += last_request_latency
+        self.current_moving_sum.frame_count += 1
+        if self.current_moving_sum.frame_count == 1 and self.total_sum.frame_count == 0:
+            self.last_moving_sum.latency = last_request_latency
+            self.last_moving_sum.frame_count = 1
```

Comment:
It makes sense to set `last_moving_sum` and `total_sum` and skip `current_moving_sum` for the first frame, so `last_moving_sum` and `total_sum` will always be updated simultaneously.